### DISCUSSION TITLE
Support for a new course?

### DISCUSSION THREAD

**Discussion by mellishim** [kantord/LibreLingo#2039] [General]
Hello! I'm a linguist working with a Native American language project. Someone suggested this program, but I'm afraid I'm not super code-savvy. Is there a robust UI in place yet for creating a new course? Or could we possibly get support for some basic modules?

**Comment by kantord** [comment]
Hi!

Thanks for you inquiry @mellishim, it sounds very exciting. I would really love to see a Native American language project in LibreLingo. :rocket:  

As you asked, of course support will be available. I'm ready to have video calls trying to work together on creating a few basic modules (of course I won't know anything about the language in question :smile_cat: ) 

@RGBradley is making very good progress on the Basque course and he actually started creating this course even before LibreLingo existed. He was initially creating it as a text document, but with an app structure in mind. Which I think means that even if the technical difficulties prove too hard in the beginning, it's not hopeless to start the course development now. Perhaps we can ask him for help as well, as he is the most knowledgeable person on LibreLingo course creation right now!

Regarding the UI, we don't have a robust UI right now. It's basically my top priority now, but of course at the same time it's probably the most complex part of the project so it will take quite some time. And perhaps it also makes sense to let the course editor and the first few courses kinda evolve together, to make sure that the features of the course editor are tailored around real problems that course authors face.

**Reply by RGBradley** [reply]
hi mellishim
I would be glad to help. As stated i laid out the Basque course in a text file first, and got some feedback, then started creating the pseudo-coding. As i have gone along, I have been constantly updating the original document file, as figure out better ways to present the grammar and vocabulary, and get corrections
you can see the latest version of my text file at https://docs.google.com/document/d/1Nxq8TCkQyqFwDGdjdrTXUf4cC-WTUBdu/edit?usp=sharing&ouid=105137969978504973552&rtpof=true&sd=true
and there is an email link you can use to contact me.
my github repository is at
https://github.com/RGBradley/LibreLingo-EU-from-EN.git
in it you can see the different units and under the skill directories, the md files that are the markdown text files that give the needed grammar points to the user and the yaml files that generate the web page
once you have a course laid out, we can work on how you can export it on you computer, so can find any errors before you upload it

regards
rgb

**Reply by RGBradley** [reply]
PS
I am also a linguist. :-)

**Reply by mellishim** [reply]
Thank you! I'll discuss with the project group and see if we can meet together :)

**Reply by mellishim** [reply]
@RGBradley I'd like to set up a chat if you're free, what's a good way to message you?

**Reply by RGBradley** [reply]
Hi mellishim
got you email and tried to reply, but don't know if you got it. We can talk, but you need to set up LibreLingo on your computer. which Kanford can help you with, then can export you course and run a local web page to test it out before uploading. 
regards
rgb

**Reply by spongefile** [reply]
@kantord I might be able to help with UI, just let me know.
